File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    cnt = 0
    stack = []
    for i in ingredient:
        stack.append(i)
        if stack[len(stack)-4:] == [1, 2, 3, 1]:
            cnt += 1
            for _ in range(4):
                stack.pop()
    return cnt

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.java
class Solution {
    public int solution(int[] ingredients) {
        int[] st = new int[ingredients.length];
        int idx = 0;
        int answer = 0;

        for (int ingredient : ingredients) {
            st[idx] = ingredient;
            idx += 1;

            if (idx >= 4) {
                if (st[idx - 1] == 1 && st[idx - 2] == 3 && st[idx - 3] == 2 && st[idx - 4] == 1) {
                    answer += 1;
                    idx -= 4;
                }
            }
        }

        return answer;
    }
}